spec/unit/shoulda/matchers/active_record/association_matcher_spec.rb

```diff
@@ -1199,6 +1199,409 @@ def belonging_to_non_existent_class(model_name, assoc_name, options = {})
       expect(Parent.new).to have_many(:children)
     end
 
+    context 'when the application is configured with strict_loading disabled by default' do
```

Comment:
What do you think about nesting all those specs under the same context? When trying this branch, I wrapped everything in a `context 'strict_loading'` to run the specs in isolation, it also helps to identify between all the other features presents in the `association_matcher_spec`